packages/manager/modules/pci/src/projects/project/storages/databases/database/metrics/metrics.controller.js

```diff
@@ -0,0 +1,135 @@
+import map from 'lodash/map';
+import sortBy from 'lodash/sortBy';
+import {
+  METRICS_TIME_RANGES,
+  CHART_METRICS_OPTIONS,
+  CHART_METRICS_REFRESH_INTERVAL,
+} from '../../databases.constants';
+
+export default class {
+  /* @ngInject */
+  constructor($translate, CucCloudMessage, DatabaseService, PciChartjsFactory) {
+    this.$translate = $translate;
+    this.CucCloudMessage = CucCloudMessage;
+    this.DatabaseService = DatabaseService;
+    this.PciChartjsFactory = PciChartjsFactory;
+    this.isLoading = true;
+    [this.selectedTimeRange] = METRICS_TIME_RANGES;
+    this.timeRanges = METRICS_TIME_RANGES;
+    this.metricsData = {};
+    this.autoRefresh = false;
+  }
+
+  $onInit() {
+    this.messageContainer =
+      'pci.projects.project.storages.databases.dashboard.metrics';
+    this.loadMessages();
+    this.trackDatabases('dashboard::metrics', 'page');
+
+    this.isLoading = false;
+
+    this.selectedTimeRange.label = this.$translate.instant(
+      `pci_databases_metrics_range_label_${this.selectedTimeRange.label_key}`,
+    );
+
+    for (let i = 0; i < this.timeRanges.length; i += 1) {
+      this.timeRanges[i].label = this.$translate.instant(
+        `pci_databases_metrics_range_label_${this.timeRanges[i].label_key}`,
+      );
+    }
+
+    this.initMetrics();
+
+    this.getMetrics();
+  }
+
+  $onDestroy() {
+    clearInterval(this.interval);
+  }
+
+  loadMessages() {
+    this.CucCloudMessage.unSubscribe(this.messageContainer);
+    this.messageHandler = this.CucCloudMessage.subscribe(
+      this.messageContainer,
+      { onMessage: () => this.refreshMessages() },
+    );
+  }
+
+  initMetrics() {
+    for (let i = 0; i < this.availableMetrics.length; i += 1) {
+      this.metricsData[this.availableMetrics[i]] = {};
+      this.metricsData[
+        this.availableMetrics[i]
+      ].chart = new this.PciChartjsFactory(
+        angular.copy(CHART_METRICS_OPTIONS.chart),
+      );
+
+      this.metricsData[this.availableMetrics[i]].chart.setTitle(
+        this.availableMetrics[i],
+      );
+
+      this.metricsData[
+        this.availableMetrics[i]
+      ].chart.setTooltipCallback('label', (item) =>
+        parseFloat(item.value, 10).toFixed(2),
+      );
+    }
+  }
+
+  onTimeRangeChange(selectedTimeRange) {
+    this.selectedTimeRange = selectedTimeRange;
+    this.trackDatabases(
+      `dashboard::metrics::display_granularity_${this.selectedTimeRange.value}`,
+      'page',
+    );
+    this.getMetrics();
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  getMetrics() {
+    for (let i = 0; i < this.availableMetrics.length; i += 1) {
```

Comment:
Same here please prefer using a `map` if you need to build an array from another one

Reply:
Should be better